modes/parkour/moderation.lua

```diff
@@ -0,0 +1,447 @@
+		return
+	end
+
+	if ban then
+		if data.banned and (data.banned == 2 or os.time() < data.banned) then tfm.exec.chatMessage("<v>[#] <r>"..player.. " has already banned.", requester) return end
+			data.bancount = data.bancount + 1
+			if data.bancount < 4 then
+				data.banned = os.time()+tonumber((data.bancount)*1440)*60*1000
+				schedule("ban_change", data.playerid, data.banned)
+				tfm.exec.chatMessage("<v>[#] <r>"..player.. " has banned for " ..((data.bancount)*1440).. " minutes.", requester)
+			else
+				schedule("ban_change", data.playerid, 1)
+				data.banned = 2
+				tfm.exec.chatMessage("<v>[#] <r>"..player.. " has permbanned.", requester)
+			end
+		system.savePlayerData(player, json.encode(data))
+	else
+		if data.banned and (data.banned == 2 or os.time() < data.banned) then 
+			data.banned = nil
+			schedule("ban_change", data.playerid, 0)
+			tfm.exec.chatMessage("<v>[#] <r>"..player.. " has unbanned.", requester)
+
+			data.bancount = data.bancount - 1
+			system.savePlayerData(player, json.encode(data))
+		else
+			tfm.exec.chatMessage("<v>[#] <r>"..player.. " not banned (yet).", requester) 
+			return 
+		end
+	end
+end
+
+local function checkBanRequest(player, data)
+	local fetch = ban_request[player]
+	if fetch then
+
+		if fetch[3] == "bancount" then
+			tfm.exec.chatMessage("<v>[#] <r>"..player.. " have " ..data.bancount.. " bans in record.", fetch[1]) 
+			ban_request[player] = nil
+			return
+		end
+
+		if fetch[3] == "resetbancount" then
+			data.bancount = 0
+			system.savePlayerData(player, data)
+			tfm.exec.chatMessage("<v>[#] <r>"..player.. "'s bans has been reset", fetch[1]) 
+			ban_request[player] = nil
+			return
+		end
+
+		local isban = fetch[3]
+		local requester = fetch[1]
+		if isban then
+			banPlayer(player, data, requester, true)
+		else
+			banPlayer(player, data, requester, false)
+		end
+		ban_request[player] = nil
+	end
+end
+
+function showChatCommands(p, command, args)
+	local commandtext = table.concat(args, " ")
+	for playername, player in pairs(tfm.get.room.playerList) do
+		if ranks.admin[playername] or ranks.mod[playername] then
+			tfm.exec.chatMessage("<BL>Ξ ["..p.."]<N2> !"..command.." "..commandtext, playername)
+		end
+	end
+end
+
+onEvent("ParsedChatCommand", function(player, cmd, quantity, args)
+	local max_args = quantity
+	if cmd == "ban" or cmd == "unban" then -- for mods / !ban Username and !unban Username 
+		if not ranks.admin[player] and not perms[player].ban then return end
+
+		if quantity < 1 then 
+			return translatedChatMessage("invalid_syntax", player)
+		end
+		
+		requestplayer = capitalize(args[1])
+		if not string.find(requestplayer, "#", 1, true) then
+			requestplayer = requestplayer .. "#0000"
+		end
+
+		showChatCommands(player, cmd, args)
+
+		if players_file[args[1]] then 
+			banPlayer(args[1], players_file[args[1]],player, cmd == "ban")
+		else
+			ban_request[args[1]] = {player, os.time() + 1000, cmd == "ban"}
+			system.loadPlayerData(args[1])
```

Comment:
I recommend adding a ban-by-id command that will just write the ban to the file; for whenever the player is offline.